snetd/cmd/free_call_users.go

```diff
@@ -0,0 +1,176 @@
+package cmd
+
+import (
+	"fmt"
+	"github.com/singnet/snet-daemon/config"
+	"github.com/singnet/snet-daemon/escrow"
+	"github.com/spf13/cobra"
+)
+
+var FreeCallUserUnLockCmd = &cobra.Command{
+	Use:   "unlock",
+	Short: "Unlock a free call user",
+	Long: "allows us to perform operations on free call users with given user_id(email_id)." +
+		" User can use 'snetd freecall unlock -u {userId}' command to unlock the User manually.",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return RunAndCleanup(cmd, args, newFreeCallUserUnLockCommandCommand)
+	},
+}
+
+var FreeCallUserResetCmd = &cobra.Command{
+	Use:   "reset",
+	Short: "Reset the count on free calls used for the given user to zero",
+	Long:  "User can use 'snetd freecall reset -u {user-id}' command to reset the free call used of this User manually.",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return RunAndCleanup(cmd, args, newFreeCallResetCountCommand)
+	},
+}
+
+//Free call user unlock command
+type freeCallUserUnLockCommand struct {
+	lockStorage *escrow.PrefixedAtomicStorage
+	userId      string
+	component   *Components
+}
+
+//Free call user unlock command
+type freeCallUserResetCountCommand struct {
+	lockStorage *escrow.PrefixedAtomicStorage
+	userStorage *escrow.FreeCallUserStorage
+	userId      string
+	component   *Components
+}
+
+// initializes and returns the new unlock user of free calls command object
+func newFreeCallUserUnLockCommandCommand(cmd *cobra.Command, args []string, pComponents *Components) (command Command, err error) {
+	userID, err := getUserId(cmd)
+	if err != nil {
+		return
+	}
+	command = &freeCallUserUnLockCommand{
+		lockStorage: pComponents.LockerStorage(),
```

Comment:
As far as I understand `FreeCallUserService` uses another lock storage: https://github.com/singnet/snet-daemon/blob/06bb92bb0ffa4811cbac50e1809f5f391657f62f/snetd/cmd/components.go#L258

Reply:
I would suggest adding new component `FreeCallLockStorage` and use it here and in `components.FreeCallUserService` method